platforms/android/tangram/src/main/cpp/androidPlatform.cpp

```diff
@@ -79,12 +86,13 @@ void AndroidPlatform::setupJniEnv(JNIEnv* jniEnv) {
     labelPickCallbackMID = jniEnv->GetMethodID(tangramClass, "labelPickCallback", "(Ljava/util/Map;FFIDD)V");
     markerPickCallbackMID = jniEnv->GetMethodID(tangramClass, "markerPickCallback", "(JFFDD)V");
 
-    if (hashmapClass) {
-        jniEnv->DeleteGlobalRef(hashmapClass);
-    }
+    // We need a reference to the class object later to invoke the constructor. FindClass produces a
+    // local reference that may not be valid later, so create a global reference to the class.
     hashmapClass = (jclass)jniEnv->NewGlobalRef(jniEnv->FindClass("java/util/HashMap"));
     hashmapInitMID = jniEnv->GetMethodID(hashmapClass, "<init>", "()V");
     hashmapPutMID = jniEnv->GetMethodID(hashmapClass, "put", "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
+
+    return JNI_VERSION_1_6;
```

Comment:
Should probably document the link here: https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/invocation.html#JNJI_OnLoad

The return value has special logic.